src/core/blocks/base_block.c

```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "base_block.h"
+
+char * ser_BlockHeader(BlockHeader *block_header, char* dest){
+    memcpy(dest, &(block_header->timestamp), sizeof(block_header->timestamp));
+
+    char * all_tx = dest+sizeof(block_header->timestamp);
+    memcpy(all_tx, &(block_header->all_tx), sizeof(block_header->all_tx));
+
+    char * prev_header_hash = all_tx+sizeof(block_header->all_tx);
+    memcpy(prev_header_hash, &(block_header->prev_header_hash), sizeof(block_header->prev_header_hash));
+
+    char * nonce = prev_header_hash+sizeof(block_header->prev_header_hash);
+    memcpy(prev_header_hash, &(block_header->nonce), sizeof(block_header->nonce));
+
+    char* terminate = nonce+sizeof(block_header->nonce);
+    return terminate;
+}
+
+char * ser_BlockHeader_alloc(BlockHeader *block_header){
```

Comment:
Can this function use `ser_BlockHeader()` to reduce duplication?

Reply:
It certainly can! I literally I was like this is a good reason to write this when I was first thinking about it